homeassistant/components/dht/sensor.py

```diff
@@ -160,18 +161,27 @@ def update(self):
 class DHTClient:
     """Get the latest data from the DHT sensor."""
 
-    def __init__(self, adafruit_dht, sensor, pin):
+    def __init__(self, sensor, pin):
         """Initialize the sensor."""
-        self.adafruit_dht = adafruit_dht
         self.sensor = sensor
-        self.pin = pin
+        self.pin = getattr(board, pin)
         self.data = {}
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Get the latest data the DHT sensor."""
-        humidity, temperature = self.adafruit_dht.read_retry(self.sensor, self.pin)
-        if temperature:
-            self.data[SENSOR_TEMPERATURE] = temperature
-        if humidity:
-            self.data[SENSOR_HUMIDITY] = humidity
+        dht = self.sensor(self.pin)
+
+        try:
+            humidity = dht.humidity
+            temperature = dht.temperature
+            if temperature:
+                self.data[SENSOR_TEMPERATURE] = temperature
+            if humidity:
+                self.data[SENSOR_HUMIDITY] = humidity
+        except RuntimeError as error:
+            pass
+        except Exception as ex:
+            _LOGGER.debug("Error returned from DHT sensor: %s", ex)
```

Comment:
Why is this just interesting at debug level?

Reply:
Should I change to warning instead? I wasn't sure if it really needed to be logged at all. The previous version just ignores the errors.

Reply:
I am not sure what to do regarding this since it generates an error:
```
Warning: homeassistant/components/dht/sensor.py:193:15: W0703: Catching too general exception Exception (broad-except)
Error: Process completed with exit code 4.
```